src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -912,6 +914,32 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         _terminal->SetSelectionAnchor(position);
     }
 
+    Core::Point ControlCore::SelectionAnchor() const
+    {
+        auto lock = _terminal->LockForReading();

```

Comment:
seems a little weird to me that these locks are taken at the ControlCore level, and not in the Terminal implementation...

Reply:
yeah, at one point, I think it was a mix between ControlCore and Terminal. I think we decided to move them up to ControlCore so that we avoid any deadlock situations. So, at least we're consistent here haha.